diff --git b/stm32mp157c-odyssey-u-boot.dtsi.orig a/stm32mp157c-odyssey-u-boot.dtsi
index abceba5..568ba3c 100644
--- b/arch/arm/dts/stm32mp157c-odyssey-u-boot.dtsi
+++ a/arch/arm/dts/stm32mp157c-odyssey-u-boot.dtsi
@@ -3,17 +3,23 @@
  * Copyright (C) 2020 Marcin Sloniewski <marcin.sloniewski@gmail.com>
  */
 
- #include "stm32mp157c-odyssey-som-u-boot.dtsi"
+#include "stm32mp157c-odyssey-som-u-boot.dtsi"
 
 / {
 	aliases {
-		mmc0 = &sdmmc1;
+        i2c1 = &i2c2;
+		mmc1 = &sdmmc2;
 		usb0 = &usbotg_hs;
 	};
+
 	config {
-		u-boot,boot-led = "heartbeat";
+        u-boot,boot-led = "heartbeat";
 		u-boot,error-led = "error";
+        u-boot,mmc-env-partition = "fip";
+		st,fastboot-gpios = <&gpiog 3 GPIO_ACTIVE_LOW>;
+		st,stm32prog-gpios = <&gpioa 13 GPIO_ACTIVE_LOW>;
 	};
+
 	led {
 		red {
 			label = "error";
@@ -28,11 +34,11 @@
 	status = "okay";
 };
 
-&sdmmc1 {
+&sdmmc2 {
 	u-boot,dm-spl;
 };
 
-&sdmmc1_b4_pins_a {
+&sdmmc2_b4_pins_a {
 	u-boot,dm-spl;
 	pins1 {
 		u-boot,dm-spl;
@@ -42,6 +48,13 @@
 	};
 };
 
+&sdmmc2_d47_pins_a {
+	u-boot,dm-spl;
+	pins {
+		u-boot,dm-spl;
+	};
+};
+
 &uart4 {
 	u-boot,dm-pre-reloc;
 };
@@ -53,6 +66,11 @@
 	};
 	pins2 {
 		u-boot,dm-pre-reloc;
+        bias-pull-up;
 	};
 };
 
+&usbotg_hs {                   
+    u-boot,force-b-session-valid;  
+};
+
