diff --git b/stm32mp157a-ed1-u-boot.dtsi.orig a/stm32mp157a-ed1-u-boot.dtsi
index 5b9772a..48d33c8 100644
--- b/arch/arm/dts/stm32mp157a-ed1-u-boot.dtsi
+++ a/arch/arm/dts/stm32mp157a-ed1-u-boot.dtsi
@@ -5,19 +5,19 @@
 
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
 #include "stm32mp15-u-boot.dtsi"
-#include "stm32mp15-ddr3-2x4Gb-1066-binG.dtsi"
+#include "stm32mp15-ddr3-1x4Gb-1066-binG.dtsi"
 
 / {
 	aliases {
-		i2c3 = &i2c4;
-		mmc0 = &sdmmc1;
+		i2c1 = &i2c2;
 		mmc1 = &sdmmc2;
+        usb0 = &usbotg_hs;
 	};
 
 	config {
 		u-boot,boot-led = "heartbeat";
 		u-boot,error-led = "error";
-		u-boot,mmc-env-partition = "fip";
+		u-boot,mmc-env-partition = "ssbl";
 		st,fastboot-gpios = <&gpioa 13 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
 		st,stm32prog-gpios = <&gpioa 14 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
 	};
@@ -31,7 +31,6 @@
 		};
 	};
 
-#ifdef CONFIG_STM32MP15x_STM32IMAGE
 	config {
 		u-boot,mmc-env-partition = "ssbl";
 	};
@@ -50,19 +49,17 @@
 			no-map;
 		};
 	};
-#endif
 };
 
-#ifndef CONFIG_TFABOOT
 &clk_hse {
 	st,digbypass;
 };
 
-&i2c4 {
+&i2c2 {
 	u-boot,dm-pre-reloc;
 };
 
-&i2c4_pins_a {
+&i2c2_pins_a {
 	u-boot,dm-pre-reloc;
 	pins {
 		u-boot,dm-pre-reloc;
@@ -165,30 +162,6 @@
 	};
 };
 
-&sdmmc1 {
-	u-boot,dm-spl;
-};
-
-&sdmmc1_b4_pins_a {
-	u-boot,dm-spl;
-	pins1 {
-		u-boot,dm-spl;
-	};
-	pins2 {
-		u-boot,dm-spl;
-	};
-};
-
-&sdmmc1_dir_pins_a {
-	u-boot,dm-spl;
-	pins1 {
-		u-boot,dm-spl;
-	};
-	pins2 {
-		u-boot,dm-spl;
-	};
-};
-
 &sdmmc2 {
 	u-boot,dm-spl;
 };
@@ -209,7 +182,6 @@
 		u-boot,dm-spl;
 	};
 };
-#endif
 
 &uart4 {
 	u-boot,dm-pre-reloc;
@@ -226,3 +198,7 @@
 		bias-pull-up;
 	};
 };
+
+&usbotg_hs {                   
+    u-boot,force-b-session-valid;  
+};
