diff --git b/stm32mp1_shared_resources.c.orig a/stm32mp1_shared_resources.c
index 208e34a..246fa6c 100644
--- b/plat/st/stm32mp1/stm32mp1_shared_resources.c
+++ a/plat/st/stm32mp1/stm32mp1_shared_resources.c
@@ -285,6 +285,7 @@ static void register_periph_iomem(uintptr_t base, unsigned int state)
 	case UART7_BASE:
 	case UART8_BASE:
 	case IWDG2_BASE:
+    case I2C2_BASE:
 		/* Allow drivers to register some non-secure resources */
 		VERBOSE("IO for non-secure resource 0x%x\n",
 			(unsigned int)base);
