diff --git b/stm32mp1_clk.c.orig a/stm32mp1_clk.c
index 73c9b14..cd12cef 100644
--- b/drivers/st/clk/stm32mp1_clk.c
+++ a/drivers/st/clk/stm32mp1_clk.c
@@ -418,7 +418,7 @@ static const struct stm32mp1_clk_gate stm32mp1_clk_gate[] = {
 	_CLK_SC_FIXED(SEC, RCC_MP_APB5ENSETR, 15, IWDG1, _PCLK5),
 	_CLK_SC_FIXED(SEC, RCC_MP_APB5ENSETR, 16, BSEC, _PCLK5),
 	_CLK_SC_SELEC(SEC, RCC_MP_APB5ENSETR, 20, STGEN_K, _STGEN_SEL),
-
+    _CLK_SC_SELEC(N_S, RCC_MP_APB1ENSETR, 22, I2C2_K, _I2C12_SEL),
 	_CLK_SELEC(SEC, RCC_BDCR, 20, RTC, _RTC_SEL),
 
 #if defined(IMAGE_BL32)
